src/index.js

```diff
@@ -20,10 +20,185 @@
 
 import SpanContext from './span_context';
 import Span from './span';
+import ConstSampler from './samplers/const_sampler';
+import InMemoryReporter from './reporters/in_memory_reporter';
+import NoopReporter from './reporters/noop_reporter';
+import ProbabilisticSampler from './samplers/probabilistic_sampler';
+import RatelimitingSampler from './samplers/ratelimiting_sampler';
+import RemoteReporter from './reporters/remote_reporter';
+import RemoteSampler from './samplers/remote_sampler';
 import Tracer from './tracer';
+import UDPSender from './reporters/udp_sender';
+import {Validator} from 'jsonschema';
+
+
+let configSchema = {
+    'type': 'object',
+    'required': ['jaeger'],
+    'properties': {
+        'jaeger': {
+            'type': 'object',
+            '$ref': '/jaeger'
+        }
+    },
+};
+
+let constSamplerSchema = {
+    'id': '/const',
+    'type': 'object',
+    'properties': {
+        'type': {'type': 'string' },
+        'param': {'type': 'boolean' }
+    },
+    'required': ['type', 'param'],
+    'additionalProperties': false
+};
+
+let probabilisticSamplerSchema = {
+    'id': '/probabilistic',
+    'type': 'object',
+    'properties': {
+        'type': {'type': 'string' },
+        'param': {'type': 'number' }
+    },
+    'required': ['type', 'param'],
+    'additionalProperties': false
+};
+
+let ratelimitingSamplerSchema = {
+    'id': '/ratelimiting',
+    'type': 'object',
+    'properties': {
+        'type': {'type': 'string' },
+        'param': {'type': 'number' }
+    },
+    'required': ['type', 'param'],
+    'additionalProperties': false
+};
+
+let remoteSamplerSchema = {
+    'id': '/remote',
+    'type': 'object',
+    'properties': {
+        'type': {'type': 'string' },
+        'param': {'type': 'string' },
+        'managerHostPort': {'type': 'string'}
+    },
+    'required': ['type', 'param'],
+    'additionalProperties': false
+};
+
+let jaegerSchema = {
+    'id': '/jaeger',
+    'type': 'object',
+    'required': ['serviceName'],
+    'properties': {
+        'serviceName': {'type': 'string'},
+        'disable': {'type': 'boolean'},
+        'sampler': {
+            'type': 'object',
+            // should be 'oneOf', but that doesn't work properly with this implementation.
+            'anyOf': [
+                {'$ref': 'const'},
+                {'$ref': 'probabilistic'},
+                {'$ref': 'ratelimiting'},
+                {'$ref': 'remote'}
+            ]
+        },
+        'reporter': {
+            'properties': {
+                'logSpans': {'type': 'boolean'},
+                'agentHost': {'type': 'string'},
+                'agentPort': {'type': 'number'},
+                'flushIntervalMs': {'type': 'number'}
+            },
+            'additionalProperties': false
+        }
+    }
+};
+
+let getSampler = (options) => {
```

Comment:
can we create a Configuration class instead and encapsulate all methods in there?